src/coreclr/jit/emitriscv64.cpp

```diff
@@ -4160,7 +4160,7 @@ regNumber emitter::emitInsTernary(instruction ins, emitAttr attr, GenTree* dst,
                     }
                     else
                     {
-                        emitIns_R_R_R(INS_mulhu, attr, codeGen->rsGetRsvdReg(), src1->GetRegNum(), src2->GetRegNum());
+                        emitIns_R_R_R(INS_mulh, attr, codeGen->rsGetRsvdReg(), src1->GetRegNum(), src2->GetRegNum());
```

Comment:
Hm, why `mulh` is used there and above? Yes it's under `needCheckOv` branch and doesn't matter which part of result we load (upper or lower), but I think `mul` is more correct.

Reply:
You can find actual multiplication in the below.
```
// n * n bytes will store n bytes result
emitIns_R_R_R(ins, attr, dst->GetRegNum(), src1->GetRegNum(), src2->GetRegNum());
```

`mulh` is used for overflow check.
```
regNumber tmpReg = dst->GetSingleTempReg();
assert(tmpReg != dst->GetRegNum());
assert(tmpReg != src1->GetRegNum());
assert(tmpReg != src2->GetRegNum());
size_t imm = (EA_SIZE(attr) == EA_8BYTE) ? 63 : 31;
emitIns_R_R_I(EA_SIZE(attr) == EA_8BYTE ? INS_srai : INS_sraiw, attr, tmpReg, dst->GetRegNum(),
              imm);
codeGen->genJumpToThrowHlpBlk_la(SCK_OVERFLOW, INS_bne, codeGen->rsGetRsvdReg(), nullptr, tmpReg);
```